platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ConsensusSnapshot.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.consensus;
+
+import com.swirlds.base.utility.ToStringBuilder;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.platform.state.MinGenInfo;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ * A snapshot of consensus at a particular round. This is all the information (except events)
+ * consensus needs to continue from a particular point. Apart from this record, consensus needs all
+ * non-ancient events to continue.
+ */
+public class ConsensusSnapshot implements SelfSerializable {
+    private static final long CLASS_ID = 0xe9563ac8048b7abcL;
+    private static final int MAX_JUDGES = 1000;
+
+    private long round;
+    private List<Hash> judgeHashes;
+    private List<MinGenInfo> minGens;
+    private long nextConsensusNumber;
+    private Instant consensusTimestamp;
+
+    private static final class ClassVersion {
+        public static final int ORIGINAL = 1;
+    }
+
+    public ConsensusSnapshot() {}
+
+    /**
+     * @param round
+     * 		the latest round for which fame has been decided
+     * @param judgeHashes
+     * 		the hashes of all the judges for this round, ordered by their creator ID
+     * @param minGens
+     * 		the round generation numbers for all non-ancient rounds
+     * @param nextConsensusNumber
+     * 		the consensus order of the next event that will reach consensus
+     * @param consensusTimestamp
+     * 		the consensus time of this snapshot
+     */
+    public ConsensusSnapshot(
+            long round,
+            @NonNull List<Hash> judgeHashes,
+            @NonNull List<MinGenInfo> minGens,
+            long nextConsensusNumber,
+            @NonNull Instant consensusTimestamp) {
+        this.round = round;
+        this.judgeHashes = judgeHashes;
```

Comment:
The `@NonNull` elements are not checked with `Objects.requireNonNull`

Reply:
will move it to big PR